analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -169,9 +168,20 @@ public void perform() {
 
                 logger.info("Completed processing for alignment and writing {} ", cache.toString());
 
-                final String newProjectName = writeProjectNameIfNeeded();
-                if ((newProjectName != null) && !newProjectName.isEmpty()) {
-                    alignmentModel.setName(newProjectName);
+                // artifactId / rootProject.getName
+                final String artifactId = SettingsFileIO.writeProjectNameIfNeeded(getProject().getRootDir());
+                if (!isEmpty(artifactId)) {
+                    logger.debug("Located artifactId ({}) for {}::{}", artifactId, alignmentModel.getGroup(),
+                            alignmentModel.getVersion());
+                    alignmentModel.setName(artifactId);
+                }
+                // groupId
+                if (isEmpty(alignmentModel.getGroup())) {
```

Comment:
It appears that even though on line 123/126 we correctly determine the groupId/artifactId e.g.
```
[AlignmentTask.java:123] Adding org.jboss.gm.common:common:1.2-SNAPSHOT to cache for scanning.
```
the project groupId does not have the correct information. While this below code works for the subset of GRPC it does _not_ work for gradle-manipulator itself. I think we may need to reconsider how we cache/etc.

Reply:
Following up on this I think it might actually be that when we create the model, the projects seem to be 'partially' defined i.e.
```
        logger.debug("Adding project {} and hashcode {} ", name, project.hashCode());
        logger.debug("Adding project with group {}", project.getGroup());
-->
[AlignmentPlugin.java:54] Adding project analyzer and hashcode 69097441 
[AlignmentPlugin.java:55] Adding project with group gradle-manipulator
```
but a bit later
```
                logger.debug("Adding {} to cache for scanning.", current);
                cache.addGAV(current);
                logger.debug("Current project {} with gav {} group is {} and {}", project.hashCode(), current, project.getGroup() , project);
-->
[AlignmentTask.java:123] Adding org.jboss.gm.analyzer:analyzer:1.2-SNAPSHOT to cache for scanning.
[AlignmentTask.java:125] Current project 69097441 with gav org.jboss.gm.analyzer:analyzer:1.2-SNAPSHOT group is org.jboss.gm.analyzer and project ':analyzer'
```

So what it *actually* looks like is that as the project gets fully defined we haven't correctly updated the model/cache.

Reply:
The groupId is now correctly defined and, if its still not set, we use the children to find a common prefix to default the root to.

Reply:
Great solution!